be/src/olap/reader.cpp

```diff
@@ -475,7 +475,7 @@ ColumnPredicate* TabletReader::_parse_to_predicate(
 }
 
 ColumnPredicate* TabletReader::_parse_to_predicate(const FunctionFilter& function_filter) {
-    int32_t index = _tablet->field_index(function_filter._col_name);
+    int32_t index = _tablet_schema->field_index(function_filter._col_name);
```

Comment:
Yes, tablet->fieled_index should not be need any more. Because we can use _tablet_schema->field_index directly. I remove the method in this pr too in case someone makes a mistake. That can cause some problems when table is enable light schema change.